test/get_test.go

```diff
@@ -0,0 +1,66 @@
+package test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+// TestGet is unit testing for the Get operation.
+func TestGet(t *testing.T) {
+	ctx, client := newKine(t)
+
+	t.Run("FailMissing", func(t *testing.T) {
+		g := NewWithT(t)
+		resp, err := client.Get(ctx, "testKey", clientv3.WithRange(""))
+
+		g.Expect(err).To(BeNil())
+		g.Expect(resp.Kvs).To(BeEmpty())
+	})
+
+	t.Run("Success", func(t *testing.T) {
+		g := NewWithT(t)
+
+		// Create a key
+		{
+			resp, err := client.Txn(ctx).
+				If(clientv3.Compare(clientv3.ModRevision("testKey"), "=", 0)).
+				Then(clientv3.OpPut("testKey", "testValue")).
+				Commit()
+			g.Expect(err).To(BeNil())
+			g.Expect(resp.Succeeded).To(BeTrue())
+		}
+
+		// Get key
+		{
+			resp, err := client.Get(ctx, "testKey", clientv3.WithRange(""))
```

Comment:
It would be nice to try to get an older version of the item. So, basically, inserting a key, updating it and deleting it. Then querying to see if it is possible to get all three versions.

Reply:
Discussed offline, adding comment here for completeness. This PR is about scaffolding the testing machinery rather than adding specific unit tests. We will deal with adding more extensive tests in followup PRs.